apisix/plugins/jwt-auth.lua

```diff
@@ -198,7 +312,11 @@ end
 
 
 local function sign_jwt_with_HS(key, auth_conf, payload)
-    local auth_secret = get_secret(auth_conf)
+    local auth_secret, err = get_secret(auth_conf)
+    if not auth_secret then
+        core.log.error("failed to sign jwt, err: ", err)
+        core.response.exit(500, "failed to sign jwt")
```

Comment:
```suggestion
        core.response.exit(503, "failed to sign jwt")
```

Use 503 for known failure.